lib/host/wasi/vinode.cpp

```diff
@@ -318,6 +319,30 @@ WasiExpect<void> VINode::pathUnlinkFile(VFS &FS, std::shared_ptr<VINode> Fd,
   return Fd->Node.pathUnlinkFile(std::string(Path));
 }
 
+WasiExpect<std::shared_ptr<VINode>>
+VINode::sockOpen(VFS &FS, __wasi_address_family_t SysDomain,
+                 __wasi_sock_type_t SockType) {
+  if (auto Res = INode::sockOpen(SysDomain, SockType); unlikely(!Res)) {
+    return WasiUnexpect(Res);
+  } else {
+    __wasi_rights_t Rights = __WASI_RIGHTS_SOCK_OPEN |
+                             __WASI_RIGHTS_SOCK_CLOSE |
+                             __WASI_RIGHTS_SOCK_SHUTDOWN;
+    return std::make_shared<VINode>(FS, std::move(*Res), Rights, Rights,
+                                    std::string(""s));
+  }
+}
+
+WasiExpect<std::shared_ptr<VINode>> VINode::sockAccept(uint16_t Port) {
+  if (auto Res = Node.sockAccept(Port); unlikely(!Res)) {
+    return WasiUnexpect(Res);
+  } else {
+    __wasi_rights_t Rights = __WASI_RIGHTS_SOCK_SHUTDOWN;
```

Comment:
Is this means that the `fd` generated by `accept` can only being shutdown, not read, write or close?